src/Core/Settings.cpp

```diff
@@ -165,4 +167,44 @@ void Settings::applyCompatibilitySetting(const String & compatibility_value)
 
 IMPLEMENT_SETTINGS_TRAITS(FormatFactorySettingsTraits, LIST_OF_ALL_FORMAT_SETTINGS)
 
+bool isChanged(const Settings & settings, size_t id)
+{
+    /// TODO: Add asserts
+    const auto & accessor = SettingsTraits::Accessor::instance();
+    return accessor.isValueChanged(settings, id);
+}
+
+/// We don't need the setting object. We just include it to make it harder to mistake different setting object types
+const std::string & getSettingName(const Settings &, size_t id)
+{
+    const auto & accessor = SettingsTraits::Accessor::instance();
+    return accessor.getName(id);
+}
+
+template <settings_supported_types T>
+T get(const Settings & settings, size_t id)
+{
+    const auto & accessor = SettingsTraits::Accessor::instance();
+    auto field = accessor.getValue(settings, id);
+    /// TODO: Add assert for debug mode
```

Comment:
`safeGet` has an assert for type safety, isn't it? Or what kind of assert would like to add?

Reply:
I didn't check safeGet. I just leave notes to myself to review it later once the API is more clear